core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -842,11 +883,15 @@ private String toStringSlow(@Nullable Channel ch, @Nullable RequestLogAccess par
                 buf.append(", sreqId=").append(sreqId);
             }
             if (ch != null) {
-                buf.append(", chanId=").append(chanId)
-                   .append(", laddr=");
-                TextFormatter.appendSocketAddress(buf, ch.localAddress());
-                buf.append(", raddr=");
-                TextFormatter.appendSocketAddress(buf, ch.remoteAddress());
+                final InetSocketAddress laddr = localAddress();
+                final InetSocketAddress raddr = remoteAddress();
+
+                buf.append(", chanId=").append(chanId);
+                buf.append(", laddr=");
+                TextFormatter.appendSocketAddress(buf, laddr);
+                if (!Objects.equals(laddr, raddr)) {
+                    buf.append(", raddr=").append(raddr);
```

Comment:
Don't we need to use `TextFormatter.appendSocketAddress()` for `raddr`?

Reply:
😱